nulib/dc-nextjs
# Refactor/experiment with chat socket component

## What does this do?
After briefly experimenting with [socket.io](https://socket.io/), looks like it needs to be implemented on both the client and server ends.  Wish I would have discovered that earlier 😀.  

The end result of this PR does:

- Re-organizes web socket connection logic/handling back into the `useChatSocket()` hook
- Provides a convenience `message` and `sendMessage` helper which a consuming component (or components) can more easily use.
- Eliminates the Chat Wrapper component we originally put in place to handle timing issues of the web socket's ready state

Visually, everything should look the same in the UI.

## How to test?  
Just a basic run through of expected functionality.  I'd try to break it, by interrupting searches, page refreshes, etc.  The connection will throw errors, but the UI should not crash.
